src/ngAnimate/animateCss.js

```diff
@@ -0,0 +1,914 @@
+
+      if (applyClassesEarly) {
+        options.$applyClasses();
+
+        // no need to calculate this anymore
+        flags.recalculateTimingStyles = false;
+
+        fullClassName = node.className + ' ' + setupClasses;
+        cacheKey = gcsHashFn(node, fullClassName);
+
+        timings = computeTimings(node, fullClassName, cacheKey);
+        maxDelay = timings.maxDelay;
+        maxDuration = timings.maxDuration;
+      }
+
+      if (maxDuration === 0 && !flags.recalculateTimingStyles) {
+        close();
+        return false;
+      }
+
+      var maxDelayTime = maxDelay * ONE_SECOND;
+      var maxDurationTime = maxDuration * ONE_SECOND;
+
+      var itemIndex = stagger
+          ? options.staggerIndex >= 0
+              ? options.staggerIndex
+              : gcsLookup.count(cacheKey) - 1
+          : 0;
+      if (!options.skipBlocking) {
+        flags.blockTransition = hasToStyles || (structural && timings.transitionDuration > 0);
+        flags.blockAnimation = timings.animationDuration > 0 &&
+                               stagger.animationDelay > 0 &&
+                               stagger.animationDuration === 0;
+      }
+
+      if (flags.blockTransition) {
+        options.$applyStyles(true, false);
+      }
+      applyBlocking(true);
+
+      return {
+        start: function() {
+          if (animationClosed) return;
+
+          runnerHost = {
+            end: endFn,
+            cancel: cancelFn
+          };
+
+          runner = new $$AnimateRunner(runnerHost);
+
+          waitUntilQuiet(start);
+
+          // we don't have access to pause/resume the animation
+          // since it hasn't run yet. AnimateRunner will therefore
+          // set noop functions for resume and pause and they will
+          // later be overridden once the animation is triggered
+          return runner;
+        },
+        end: endFn,
+        duration: maxDuration,
+        delay: maxDelay,
+        transitions: timings.transitionDuration > 0,
+        keyframes: timings.animationDuration > 0
+      };
+
+      function endFn() {
+        close();
+      }
+
+      function cancelFn() {
+        close(true);
+      }
+
+      function close(rejected) { // jshint ignore:line
+        // if the promise has been called already then we shouldn't close
+        // the animation again
+        if (animationClosed || (animationCompleted && animationPaused)) return;
+        animationClosed = true;
+        animationPaused = false;
+
+        $$jqLite.removeClass(element, setupClasses);
+        $$jqLite.removeClass(element, activeClasses);
+
+        forEach(temporaryStyles, function(entry) {
+          // There is only one way to remove inline style properties entirely from elements.
+          // By using `removeProperty` this works, but we need to convert camel-cased CSS
+          // styles down to hyphenated values.
+          node.style.removeProperty(normalizeCssProp(entry[0]));
```

Comment:
What is the problem with `node.style.someProp = ''` ?
(It seems to work on latest Chrome, Firefox, IE, unless I misunderstood the meaning of _remove inline style properties entirely_.)